src/utils.js

```diff
@@ -0,0 +1,144 @@
+const core = require('@actions/core');
+const fs = require('fs');
+const glob = require('glob');
+const xmlParser = require('fast-xml-parser');
+const he = require('he');
+
+function findTrxFiles(baseDir) {
+  core.info(`Looking for trx files in '${baseDir}'...`);
+
+  const files = glob.sync(baseDir + '/**/*.trx', {});
+  if (!files || files.length === 0) {
+    throw new RangeError('There were no trx files found.');
+  }
+
+  core.info(`The following trx files were found:`);
+  core.info(`\t${files.join('\n\t')}`);
+  return files;
+}
+
+async function transformAllTrxToJson(trxFiles) {
+  const transformedTrxReports = [];
+
+  for (const trx of trxFiles) {
+    transformedTrxReports.push(await transformTrxToJson(trx));
+  }
+
+  return transformedTrxReports;
+}
+
+async function transformTrxToJson(filePath) {
+  let trxDataWrapper;
+
+  core.info(`Transforming file ${filePath}`);
+
+  const xmlData = fs.readFileSync(filePath, 'utf-8');
+  const options = {
+    attributeNamePrefix: '_',
+    textNodeName: '#text',
+    ignoreAttributes: false,
+    ignoreNameSpace: false,
+    allowBooleanAttributes: true,
+    parseNodeValue: true,
+    parseAttributeValue: true,
+    trimValues: true,
+    cdataTagName: '__cdata',
+    cdataPositionChar: '\\c',
+    parseTrueNumberOnly: false,
+    arrayMode: false,
+    attrValueProcessor: (val, _attrName) =>
+      he.decode(val, {
+        isAttributeValue: true
+      }),
+    tagValueProcessor: (val, _tagName) => he.decode(val),
+    stopNodes: ['parse-me-as-string']
+  };
+
+  if (xmlParser.validate(xmlData.toString()) === true) {
+    const jsonString = xmlParser.parse(xmlData, options, true);
+    const testData = jsonString;
+    const runInfos = testData.TestRun.ResultSummary.RunInfos;
+    if (runInfos && runInfos.RunInfo._outcome === 'Failed') {
+      core.warning('There is trouble');
+    }
+
+    const reportHeaders = getReportHeaders(testData);
+    trxDataWrapper = {
+      TrxData: jsonString,
+      IsEmpty: testData.TestRun.TestDefinitions ? false : true,
+      ReportMetaData: {
+        TrxFilePath: filePath,
+        ReportName: `${reportHeaders.reportName}-check`,
+        ReportTitle: reportHeaders.reportTitle,
+        TrxJSonString: JSON.stringify(jsonString),
+        TrxXmlString: xmlData
+      }
+    };
+  }
+  return trxDataWrapper;
+}
+
+function getReportHeaders(data) {
+  let reportTitle = '';
+  let reportName = '';
+  const isEmpty = data && data.TestRun && data.TestRun.TestDefinitions ? false : true;
```

Comment:
Ternaries that set a value to true/false are usually easier to read if set to the condition (or the negated condition in this case). But this is another thing that isn't a huge deal.

Reply:
Oh I totally agree, it took me a minute to figure out what was going to happen when. :)  I tried to keep it close to the original code but I can probably refactor that since apparently I'm not the only one who has trouble reading nested ternaries.